scene/main/viewport.cpp

```diff
@@ -2634,6 +2634,7 @@ void Viewport::_gui_control_grab_focus(Control *p_control) {
 		return;
 	get_tree()->call_group_flags(SceneTree::GROUP_CALL_REALTIME, "_viewports", "_gui_remove_focus");
 	gui.key_focus = p_control;
+	emit_signal("gui_focus_changed", p_control);
```

Comment:
There's `NOTIFICATION_FOCUS_ENTER` sent to `p_control` already, can't you use that one instead?

Reply:
It is only sent to the control that is focused. Other controls don't get this notification and they don't know that focus changed either.